### DISCUSSION TITLE
How to switch to NET6?

### DISCUSSION THREAD

**Discussion by MikeKry** [OrchardCMS/OrchardCore#11185] [Q&A] [answered]
After updating to NET6 packages, Orchard fails to start. 

![image](https://user-images.githubusercontent.com/52829889/153828364-52ad1f43-1fe2-4692-add1-d213ecc0e524.png)
![image](https://user-images.githubusercontent.com/52829889/153828404-546d7fd8-19b0-48dc-acdf-647921729ce0.png)

Is there some action we should do after updating packages?

**Comment by hishamco** [comment]
Do you have .NET Core 6 installed?

Are you using source or NuGet packages?

**Comment by MikeKry** [comment]
@hishamco 

Yes, I have .NET 6 installed and I am using nugets 1.3.0-preview-16704

It also seems that it works under production env. (ASPNETCORE_ENVIRONMENT) just not in development

**Reply by hishamco** [reply]
Strange!! please wait for others to reply, I didn't used the latest NuGet packages.

May be it's a caching issue

**Comment by MikeKry** [comment, marked as answer]
And there was no change in project other than updating nugets & updating target framework of project to net6.0

**Reply by giannik** [reply]
https://github.com/OrchardCMS/OrchardCore/issues/11128

**Reply by MikeKry** [reply]
@giannik  @hishamco  

seems that this could be another issue, we do not have RazorRuntimeCompilation set in csproj

this is csproj def: 

![image](https://user-images.githubusercontent.com/52829889/153870618-466f4072-0ec0-49b5-b353-2f8b036a8bed.png)

**Reply by MikeKry** [reply]
Hm, it seems it is caused by "PreserveCompilationReferences", which could be related to RazorRuntimeCompilation

**Reply by MikeKry** [reply]
@hishamco 
If it is the problem and should not be used - documentation should be updated https://github.com/OrchardCMS/OrchardCore/blob/2f3b86545f87baad359622f85c19c5b7b1cf6c9e/src/docs/guides/decoupled-cms/README.md

**Reply by jtkech** [reply]
Yes PreserveCompilationReferences is no more used

https://github.com/OrchardCMS/OrchardCore/blob/1d4f2837dfcd84e3119a7ce3afe7475d0a805464/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj#L8-L14

https://github.com/OrchardCMS/OrchardCore/blob/1d4f2837dfcd84e3119a7ce3afe7475d0a805464/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj#L26-L28

**Reply by MikeKry** [reply]
@jtkech so should I create task to fix the guide for decoupled cms?

**Comment by jtkech** [comment]
Yes, would be fine to create a PR updating the related doc, thanks

**Comment by ns8482e** [comment]
My experience is that the runtime compilation is not playing well with .NET6.0.  and IMHO just disable it

**Reply by jtkech** [reply]
Anyway, don't use anymore `PreserveCompilationReferences` ;)
